tests/nim/testnim.py

```diff
@@ -0,0 +1,94 @@
+import glob
+import os
+import platform
+import shutil
+import subprocess
+from pathlib import Path
+
+test_nim_dir = Path(__file__).absolute().parent
+test_dir = test_nim_dir.parent
+generated_dir = test_nim_dir / "generated"
+root_path = test_dir.parent
+flatc_exe = Path("flatc.exe" if platform.system() == "Windows" else "flatc")
+flatc_path = root_path / flatc_exe
+assert flatc_path.exists(), "Cannot find the flatc compiler " + str(flatc_path)
+
+# Execute the flatc compiler with the specified parameters
+def flatc(options, schema, prefix=None, include=None, data=None, cwd=root_path):
```

Comment:
I can't remember if this was discussed but is there a reason that we generate code in this file instead of `/scripts/generate_code.py`? Thats where we generate code for all other languages

Reply:
No specific reason, I wasn't sure if I was expected to pollute the global directory with all the generated files so I just generated it for the tests specifically, will change.

Reply:
Yeah, I'm ambivalent as to whether we should be generating so many files and checking them in. It's noisy but sort of helpful for reviewing small generated code changes. Regardless, it is how we currently do things